### PR TITLE
SLCORE-645   Remove getServerHotspots() from SonarLintEngine in favour of RPC

### PR DESCRIPTION
**Pull request by vojtech-suchy-sonarsource**

### CODE REVIEW

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java [hunk 1]
```diff
@@ -1218,35 +1205,46 @@ void shouldContainExtendedDescription() throws Exception {
...
-          .isEqualTo(expected);
+
+      backend.getConfigurationService().didAddConfigurationScopes(new DidAddConfigurationScopesParams(
+        List.of(new ConfigurationScopeDto(CONFIG_SCOPE_ID, null, true, projectName, new BindingConfigurationDto(CONNECTION_ID, projectKey, true)))));
+
+      await().untilAsserted(() -> assertThat(didSynchronizeConfigurationScopes).isNotEmpty());
+
+      var ruleDetailsResponse = backend.getRulesService().getEffectiveRuleDetails(new GetEffectiveRuleDetailsParams(CONFIG_SCOPE_ID, javaRuleKey("S106"), null)).get();
+      var ruleTabs = ruleDetailsResponse.details().getDescription().getRight().getTabs();
+      assertThat(ruleTabs.get(ruleTabs.size() -1).getContent().getLeft().getHtmlContent()).contains(expected);
```
Comment (henryju) [inline its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java]: Code style issue (missing space before "1")

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java [hunk 2]
```diff
@@ -1449,6 +1443,11 @@ public void didChangeTaintVulnerabilities(String configurationScopeId, Set<UUID>
         didChangeTaintVulnerabilitiesEvents
           .add(new DidChangeTaintVulnerabilitiesParams(configurationScopeId, closedTaintVulnerabilityIds, addedTaintVulnerabilities, updatedTaintVulnerabilities));
       }
+
+      @Override
+      public void didSynchronizeConfigurationScopes(DidSynchronizeConfigurationScopeParams params) {
+        didSynchronizeConfigurationScopes.add(params.getConfigurationScopeIds().toString());
```
Comment (henryju) [inline its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:1451]: This look suspicious. Shouldn't it be 
`didSynchronizeConfigurationScopes.addAll(params.getConfigurationScopeIds());`

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java [hunk 3]
```diff
@@ -1218,35 +1205,46 @@ void shouldContainExtendedDescription() throws Exception {
...
         expected = "<h1>Title\n</h1><strong>my dummy extended description</strong>";
       }
-      assertThat(engine.getActiveRuleDetails(endpointParams(ORCHESTRATOR), serverLauncher.getJavaImpl().getHttpClient(CONNECTION_ID), javaRuleKey("S106"), projectKey).get()
-        .getExtendedDescription())
-          .isEqualTo(expected);
+
+      backend.getConfigurationService().didAddConfigurationScopes(new DidAddConfigurationScopesParams(
+        List.of(new ConfigurationScopeDto(CONFIG_SCOPE_ID, null, true, projectName, new BindingConfigurationDto(CONNECTION_ID, projectKey, true)))));
+
+      await().untilAsserted(() -> assertThat(didSynchronizeConfigurationScopes).isNotEmpty());
```
Comment (henryju) [inline its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:1208]: There is little chance of issue here, but you could wait for the exact config scope:
`await().untilAsserted(() -> assertThat(didSynchronizeConfigurationScopes).contains(CONFIG_SCOPE_ID));`

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java [hunk 4]
```diff
@@ -1212,8 +1207,14 @@ void shouldContainExtendedDescription() throws Exception {
       await().untilAsserted(() -> assertThat(didSynchronizeConfigurationScopes).isNotEmpty());
 
       var ruleDetailsResponse = backend.getRulesService().getEffectiveRuleDetails(new GetEffectiveRuleDetailsParams(CONFIG_SCOPE_ID, javaRuleKey("S106"), null)).get();
-      var ruleTabs = ruleDetailsResponse.details().getDescription().getRight().getTabs();
-      assertThat(ruleTabs.get(ruleTabs.size() -1).getContent().getLeft().getHtmlContent()).contains(expected);
+      var ruleDescription = ruleDetailsResponse.details().getDescription();
+      if (!ORCHESTRATOR.getServer().version().isGreaterThanOrEquals(9, 5)) {
```
Comment (henryju) [inline its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:1212]: Minor comment, but I find this kind of double negation difficult to read. Maybe invert the if/else condition?

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java [hunk 5]
```diff
@@ -1168,9 +1168,10 @@ void shouldFailIfNotAuthenticated() {
 
       adminWsClient.settings().set(new SetRequest().setKey("sonar.forceAuthentication").setValue("true"));
       try {
+        var rule = ORCHESTRATOR.getServer().version().isGreaterThan(7, 9) ? "'java:S106'" : "'squid:S106'";
```
Comment (henryju) [inline its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:1171]: you have the method `javaRuleKey("S106")` just doing the same a few line below

**Review by henryju — CHANGES_REQUESTED** [review]

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java (hunk 2 above)
Comment (vojtech-suchy-sonarsource) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:1451]: well spotted, that was leftover from a quick attack on the problem

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java (hunk 1 above)
Comment (vojtech-suchy-sonarsource) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java]: Addressed in 48b21ba

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java (hunk 3 above)
Comment (vojtech-suchy-sonarsource) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java]: Addressed in 48b21ba

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java (hunk 4 above)
Comment (vojtech-suchy-sonarsource) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java]: Addressed in 48b21ba

**Review by henryju — APPROVED** [review]

**Pull request merged**